design-proposals/network-binding-plugin/network-binding-plugin.md

```diff
@@ -941,3 +984,151 @@ Libvirt configuration snippet:
     <rom enabled='no'/>
 </interface>
 ```
+
+## Appendix F: Concrete example for vDPA binding
+
+Present a concrete example of the actions needed at each integration
+point, considering the option suggested in the
+[design section](#design-details).
+
+### Kubevirt CR Specification
+
+The vDPA binding is registered to the Kubevirt CR.
+The plugin is then referenced by name from the VM Specification.
+
+> **Note**: The vDPA binding does not need to perform changes in the
+> pod network, therefore, there is no need for a
+> network-attachment-definition.
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    network:
+      binding:
+        vdpa:
+          sidecarImage: quay.io/kubevirt/network-vdpa-binding
+          downwardAPI: device-info
+```
+
+### VM Specification
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vmi-1-bridge
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+            - name: blue
+              mac: 12:34:56:78:9a:bc
+              binding:
+                name: vdpa
+      networks:
+        - name: blue
+          multus:
+            networkName: vdpa-network
+```
+
+### NetworkAttachmentDefinition Specification
+
+Not required for vDPA binding
+
+> **Note**: Just to clarify, a NAD is required for the vDPA CNI and
+> device plugin, but it is not in the scope of this doc. An extra NAD
+> for
+> the network binding plugin is not required.
+### Webhook Admitter
+
+- The core KubeVirt admitter assures that only one method of binding
+  definition is possible: Either the old style method or the new one.
+- No addmiter is provided as part of the vDPA binding.
+
+### Pod Definition
+
+The VMI controller updates the pod manifest per the VMI specification.
+
+- From the VMI spec, the following snippet is detected:
+
+```yaml
+name: blue
+binding:
+  name: vdpa
+```
+
+- As a result, the pod is updated with the following annotations & spec:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: virt-launcher-123
+  annotations:
+    kubevirt.io/network-device-info-map: '{
+      blue:
+        "device-info": {
+          "type": "vdpa",
+          "version": "1.1.0",
+          "vdpa": {
+            "parent-device": "vdpa:0000:65:00.3",
+            "driver": "vhost",
+            "path": "/dev/vhost-vdpa-1",
+            "pci-address": "0000:65:00.3"
+          }
+        }
+    }'
+  spec:
+    containers:
+      - name: compute
+
+      - name: hook-sidecar-1
+        volumeMounts:
+          - name: network-device-info-map
+            mountPath: /etc/pod-info
```

Comment:
How about mounting the device info at its own directory? maybe `/etc/net-device-info`?

Note that the current SR-IOV mapping annotation is mounted to `/etc/pod-info`

Reply:
It is intentionally under the same dir. The path is different - /etc/pod-info/device-info

Reply:
Also commented on the impl. PR 
I would store the new device-info annotation in different dir then the sriov-network-pci one.
IMO it will be easier to debug that way, also avoid potential overwrites, if tomorrow someone changes the downward API volumes render code.